### DISCUSSION TITLE
Request: Quake-style (Visor) Overlay Terminal Mode

### DISCUSSION THREAD

**Discussion by yoshimi-I** [ghostty-org/ghostty#3733] [Feature Requests, Ideas] [answered]
# Feature Request: Quake-style (Visor) Overlay Terminal Mode

**Summary**  
I would love to have a “Quake-style” or “Visor-style” overlay mode in Ghostty. In this mode, the terminal would slide down from the top of the screen (overlaying other application windows) and hide again with a keybind. This would allow me to quickly access the terminal while keeping my current applications visible underneath.

---

## Motivation / Use Case
- Quickly reference code, logs, or commands without leaving full view of other apps.  
- Maintain focus on other windows while still having a terminal readily available.  
- Similar to iTerm2’s “Visor” feature or the “Quake-style” terminal used in many Linux environments.

---

## Proposal
1. **Configuration/Flag:**  
   Add a new configuration option (e.g., `quick-terminal-mode = "visor"`) or a CLI flag like `--overlay`.  
2. **Behavior:**  
   When enabled, pressing a global keybind (e.g., `Shift + Enter`) would animate the terminal sliding down from the top edge of the screen and remain on top of other applications.  
3. **Toggle:**  
   Pressing the same keybind again would hide the terminal back to the top.  
4. **Optional Opacity:**  
   Ideally, allow transparency so the user can see underlying apps if desired.

---

## Additional Details
- The current `quick-terminal` feature is close, but it behaves as a normal window rather than truly overlaying.  
- This feature would mimic a “visor” effect, letting the user see and interact with other apps behind the translucent terminal.  
- Some macOS permissions (Accessibility) may be required for global overlays.  
- Could be macOS-only initially or made cross-platform if feasible.

---

## Benefits
- **Enhanced Productivity:** Useful for developers who rapidly switch between code editors, browsers, and terminals.  
- **Reduced Friction:** Eliminates constant window arrangement or switching for quick terminal commands.

---

## Thank You
Adding an overlay-style terminal would make it even more powerful for everyday use.

**Comment by yankcrime** [comment, marked as answer]
This already exists, see https://ghostty.org/docs/config/keybind/reference#toggle_quick_terminal

**Reply by mitchellh** [reply]
Note this is macOS only currently but that's because we can't figure out a good way [yet] to do this on Linux with GTK.

**Reply by flying-sheep** [reply]
Hm, is GTK a good path? I feel like it’s become more and more stripped down to serve GNOME alone. Maybe Qt is the better bet for Linux.

**Reply by JPnux** [reply]
> Note this is macOS only currently but that's because we can't figure out a good way [yet] to do this on Linux with GTK.

Tilix works with "quake mode" under Gnome. How do they do it ?

**Reply by gauravkumar37** [reply]
yakuake also works in QT

**Reply by pluiedev** [reply]
FYI we did figure out a way to do this with GTK 4, which should work on any Wayland platform except GNOME (as they don't implement the necessary protocols required for quick terminal). 

Tilix, Guake, etc. are all GTK 3 terminals and rely on massive hacks on Wayland to position themselves as a floating window with a fixed position, which is not normally allowed on Wayland.

**Reply by masch** [reply]
Are you planing do it on GTK 4 soon on Wayland? It'll be great to have it.

**Reply by 00-kat** [reply]
It exists already on tip → #6027.

**Comment by mitchellh** [comment]
Issues aren't for feature requests. Converted.

**Comment by tfactor2** [comment]
Thanks for the base implementation.

Any plans to support tabs?

**Reply by connebs** [reply]
Tab support would be great! Very impressed with ghostty (been following development for a while). Tabs in the "quick terminal" is the only feature currently lacking that I need in order to daily ghostty.

**Reply by tfactor2** [reply]
tmux can be an okayish replacement though:
![image](https://github.com/user-attachments/assets/92a82bae-afd6-4e56-9771-5c42c5c05c9e)

**Reply by rhodes-b** [reply]
https://github.com/ghostty-org/ghostty/discussions/4906

**Reply by nhoss2** [reply]
+1 tabs on quick terminal is the last feature i would like before swapping

**Reply by pluiedev** [reply]
Please read the linked discussion and avoid posting bump comments. Supporting tabs would require a custom tab bar solution which has not been made yet, and you just needlessly notified maybe 10 people.
